packages/grid/x-grid/src/tests/rows.XGrid.test.tsx

```diff
@@ -358,6 +358,34 @@ describe('<XGrid /> - Rows', () => {
       expect(isVirtualized).to.equal(false);
     });
 
+    it('should set the virtual page to 0 when resetting rows to a non virtualized length', () => {
+      const { setProps } = render(<TestCaseVirtualization nbRows={996} hideFooter height={600} />);
+
+      const gridWindow = document.querySelector('.MuiDataGrid-window')!;
+      gridWindow.scrollTop = 10e6; // scroll to the bottom
+      gridWindow.dispatchEvent(new Event('scroll'));
+
+      let lastCell = document.querySelector('[role="row"]:last-child [role="cell"]:first-child')!;
+      expect(lastCell.textContent).to.equal('995');
+
+      let virtualPage = apiRef!.current!.getState().rendering!.virtualPage;
+      expect(virtualPage).to.equal(98);
+
+      setProps({ nbRows: 9 });
+
+      lastCell = document.querySelector('[role="row"]:last-child [role="cell"]:first-child')!;
+      expect(lastCell.textContent).to.equal('8');
```

Comment:
A detail, we have tried to use:

```suggestion
      expect(lastCell).to.have.text('8');
```

so far for this context, same with the other